engine/src/main/java/io/camunda/zeebe/engine/processing/timer/TriggerTimerProcessor.java

```diff
@@ -171,4 +179,19 @@ private void rescheduleTimer(
         writer,
         sideEffects::accept);
   }
+
+  private Interval refreshInterval(final long lastDueDate, final Interval interval) {
```

Comment:
:wrench: We could move the method to the `Interval` class. The purpose of the method is to create a new interval with a given start. There are similar examples in Java's date/time API.  

For example:
```
class Interval {
  public Interval withStart(final Instant start) {
    // ..
  }
}
```